libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/rest_channel_service_client_factory.py

```diff
@@ -43,16 +47,41 @@ async def create_connector_client(
             raise TypeError(
                 "RestChannelServiceClientFactory.create_connector_client: audience can't be None or Empty"
             )
+        
+        if context.activity.is_agentic_request():
 
-        token_provider: AccessTokenProviderBase = (
-            self._connection_manager.get_token_provider(claims_identity, service_url)
-            if not use_anonymous
-            else self._ANONYMOUS_TOKEN_PROVIDER
-        )
+            # breakpoint()
 
-        token = await token_provider.get_access_token(
-            audience, scopes or [f"{audience}/.default"]
-        )
+            if not context.identity:
+                raise ValueError("context.identity is required for agentic activities")
+            
+            connection = self._connection_manager.get_token_provider(context.identity, service_url)
+            if connection._msal_configuration.ALT_BLUEPRINT_ID:
+                connection = self._connection_manager.get_connection(connection._msal_configuration.ALT_BLUEPRINT_ID)
```

Comment:
Log here. Using the ALT vs Not using the ALT is going to be a bug farm. Let's at least give the logs here for us to point at for debugging.